### PR TITLE
bug fix: attribution missing authors name

### PR DESCRIPTION
**Pull request by Andrew-WMDE**
See: https://phabricator.wikimedia.org/T105803

### CODE REVIEW

**Comment by wmde-manicki** [comment]
Checked and worked, nice!

File: app/AttributionGenerator.js [hunk 1]
```diff
@@ -158,7 +158,9 @@ $.extend( AttributionGenerator.prototype, {
 		// Prefer attribution over author:
 		var $attribution = this._asset.getAttribution();
 		if( $attribution ) {
-			return format === 'html' ? $attribution : $attribution.text();
+			if( $attribution.length && $attribution.text().length ) {
```
Comment (JeroenDeDauw) [inline app/AttributionGenerator.js]: In the WB coding style guidelines we have something about not doing implicit conversions like the cast from `$attribution.text().length` to boolean. `$attribution.text().length > 0`

Comment (thiemowmde) [inline app/AttributionGenerator.js]: In JavaScript this is common, we also do this in Wikibase related JavaScript code.

Comment (JeroenDeDauw) [inline app/AttributionGenerator.js]: Sure. It being common does not mean it's good though, especially when talking about JS practices :)

In this particular case I doubt anyone would get confused, though for things less known than the length attribute, such as all own code, it does make a real difference.

Comment (thiemowmde) [inline app/AttributionGenerator.js]: Using `if ( $jQueryCollection.length )` like it is a boolean is common practice. Not so much for the text. I would prefer an `!== ''` check there.

Comment (JeroenDeDauw) [inline app/AttributionGenerator.js]: Do you disagree with me when I ask to not do this?

```js
if ( getAwesomenessCount() ) {  }
```

where `getAwesomenessCount` returns an int

Comment (wmde-manicki) [inline app/AttributionGenerator.js]: Ooops, I didn't think about it. But it does definitely make sense to be explicit there, of course.

I quickly looked at code in this repo and there are both approaches mixed. Then explicit checking the integer seems to be slightly more often used. I will make some tour through the code to make it more consistent. Thanks for pointing this out.

@Andrew-WMDE sorry for pointing you in the wrong direction!

**Pull request merged**